File: backend/routes/predict.js
const express = require("express");

const router = express.Router();

const Prediction =
    require("../models/Prediction");


// Prediction Route
router.post("/", async (req, res) => {

    try {

        const {

            confirmed,

            deaths,

            recovered,

            active,

            population

        } = req.body;

        // Basic AI Logic
        let prediction = "LOW RISK";

        if (

            confirmed > 100000 ||

            deaths > 5000 ||

            active > 50000

        ) {

            prediction = "HIGH RISK";

        }

        // Save prediction
        await Prediction.create({

            confirmed,

            deaths,

            recovered,

            active,

            population,

            result: prediction

        });

        res.json({

            prediction

        });

    } catch (error) {

        console.log(error);

        res.status(500).json({

            message:
                "Prediction Failed"

        });

    }

});

module.exports = router;

File: ai-model/predict.py
import pickle
import numpy as np

# Load model
model = pickle.load(open("outbreak_model.pkl", "rb"))

# Example input:
# Confirmed, Deaths, Recovered, Active, Population

sample_data = np.array([[200000, 5000, 150000, 45000, 1000000]])

prediction = model.predict(sample_data)

if prediction[0] == 1:
    print("HIGH RISK OUTBREAK")
else:
    print("LOW RISK")